src/sonic-yang-models/yang-templates/sonic-acl.yang.j2

```diff
@@ -75,8 +75,20 @@ module sonic-acl {
 					}
 				}
 
-				choice ip_prefix {
-
+				choice src_dst_address {
+					case l2_src_dst_address {
+						when "(/acl:sonic-acl/acl:ACL_TABLE/acl:ACL_TABLE_LIST[ACL_TABLE_NAME=current()/acl:ACL_TABLE_NAME]/acl:type = 'L2')";
+						leaf SRC_MAC {
+							type string {
+								pattern "[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}|[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}/[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}";
```

Comment:
>pattern [](http://example.com/codeflow?start=8&length=7)

Should we define a shared type, and name it like stypes:mac_and_mask?
I see the pattern repeated below.

Reply:
Added common type and used it in acl.yang